sys/mips/timer.c

```diff
@@ -85,10 +100,13 @@ static int mips_timer_stop(timer_t *tm) {
 static bintime_t mips_timer_gettime(timer_t *tm) {
   device_t *dev = tm->tm_priv;
   mips_timer_state_t *state = dev->state;
-  uint64_t count = read_count(state);
-  uint32_t sec = count / tm->tm_frequency;
-  uint32_t frac = count % tm->tm_frequency;
-  bintime_t bt = bintime_mul(HZ2BT(tm->tm_frequency), frac);
+  uint32_t sec, ticks;
+  WITH_INTR_DISABLED {
+    read_count(state);
+    sec = state->sec;
+    ticks = state->cntr_modulo;
+  }
+  bintime_t bt = bintime_mul(HZ2BT(tm->tm_frequency), ticks);
```

Comment:
`HZ2BT()` does 64-bit division, which is as expensive as modulo. One way to work around this is to compute and store it in the timer structure at initialization time.

Reply:
Missed that, thanks!